### ISSUE TITLE
.exe of Epic games being imported as image?

### ISSUE THREAD

**Issue by alpe12** [JosefNemec/Playnite#2618]
None of my Epic installed games have icons.

And if I go to Edit > Media, and try to import icon "from Play action" it says "Cannot get icon if Play action is missing or is set to URL".

Log: https://pastebin.com/c9CB0ngG

Maybe this helps:
I clicked on "Open metadata folder" and noticed that all Epic games metadata folders have the exe of the game in it.
Maybe it failed to extract the icon for some reason, and copied the exe instead. Or is this normal?
![eg](https://user-images.githubusercontent.com/8551127/139517426-3f95e6a3-a02a-4acf-9c6c-87165f943686.jpg)

Diagnostics package ID: 48524ec7-7221-4c17-82d0-4178211f24ea

**Comment by JosefNemec** [comment]
I'm not sure how this could have happened. I tried it on my system and icons are extracted properly as .ico files for me (whem metadata are downloaded automatically after importing Epic games). Are you sure you didn't use some extensions that could do it?

**Comment by JosefNemec** [comment]
Also icon from Play action button only works if you have a game with manually set Play action that point's to exe file.

**Comment by alpe12** [comment]
I tried changing to the default theme and then restarting in safe mode. Then I removed the game from library, and updated game library again. It imported the exe as before.
8c3842d4-3ed9-4615-b133-09595a756fd3

Just tried a fresh install without any modification. Same issue.
dc0e6c0f-7d61-4a75-9446-69c87c512e24

**Comment by alpe12** [comment]
I searched the error 0X88982F50 and judging from the first results it is from Windows Live Photo Gallery. I opened the Photos app on windows and a login popup appears. Maybe this is the issue? If so, wouldn't it be possible to add a library to use instead of the Windows Photos?

**Comment by JosefNemec** [comment]
We don't use Windows Photos for anything.

**Comment by alpe12** [comment]
I searched it because of this:
`[The component cannot be found. (Exception from HRESULT: 0x88982F50)](https://pastebin.com/BqrkFs6s)`

Any suggestion on things I can test to shed some light into this?

**Comment by JosefNemec** [comment]
That error is there because UI toolkit tries to display image from those exe files which will obviously fail. But the root issue is that those exe files shouldn't be there in the first place.

**Comment by JosefNemec** [comment]
Found the issue, I'll try to fix it in Epic integration plugin, fix in Playnite itself is more complicated.

**Comment by JosefNemec** [comment]
Will have to be fixed in Playnite itself, the same issue can happen with any metadata source that provides exe as icons :(

**Comment by JosefNemec** [comment]
Fixed in master.

**Comment by alpe12** [comment]
@JosefNemec
I grabbed Aven Colony from epic and updated my library on Playnite.
Then I installed it, and updated the metadata so it would import the icon. But it didn't
I looked at the media folder and there was an exe there that was added when my library was updated. BTW, I'm already using the updated Playnite version.
I removed the exe and updated metadata again. This time the exe wasn't imported. But no icon was imported either.
https://pastebin.com/P64HfmWj
